src/endpoints/pool/pool.service.ts

```diff
@@ -105,6 +109,11 @@ export class PoolService {
       transaction.receiverShard = AddressUtils.computeShard(AddressUtils.bech32Decode(transaction.receiver), shardCount);
     }
 
+    const metadata = await this.transactionActionService.getTransactionMetadata(this.poolTransactionToTransaction(transaction), false);
```

Comment:
In our Python implementation, we have a small issue with parsing some badly-formatted transactions. If the issue exists in the JS implementation, as well, a tiny failure might make the whole request fail, I think. Perhaps have a more pessimistic approach?

Reply:
Additionally, maybe we should benchmark this? Or is the logic applied for a mere page of transactions, not all?

If applied on all currently in pool, we might have a small performance issue?

:thought_balloon:

Reply:
> a tiny failure might make the whole request fail, I think
no, if the decoding fails, it will just returned `undefined`, therefore not applying the function name

regarding the performance impact, it is ok since we cache the entire pool so the processing only happens once